src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/ParenPadCheck.java

```diff
@@ -253,31 +267,10 @@ private void processExpression(DetailAST ast) {
             else if (currentNode.getType() == TokenTypes.RPAREN && !isInTypecast(currentNode)) {
                 processRight(currentNode);
             }
-            else if (currentNode.hasChildren() && !isAcceptableToken(currentNode)) {
-                // Traverse all subtree tokens which will never be configured
-                // to be launched in visitToken()
-                currentNode = currentNode.getFirstChild();
-                continue;
-            }
-
-            // Go up after processing the last child
-            while (currentNode.getNextSibling() == null && currentNode.getParent() != ast) {
-                currentNode = currentNode.getParent();
-            }
             currentNode = currentNode.getNextSibling();
         }
     }
 
-    /**
-     * Checks whether AcceptableTokens contains the given ast.
-     *
-     * @param ast the token to check.
-     * @return true if the ast is in AcceptableTokens.
-     */
-    private boolean isAcceptableToken(DetailAST ast) {
-        return acceptableTokens.get(ast.getType());
-    }
-
```

Comment:
we don't need this anymore we needed it to avoid checking the same node twice. while doing a deep scan when we found a subtree that is acceptableToken we don't visit it because it will be picked up with `visitToken()` later

now there is no deep scan for tokens so we don't need this check. I removed this method, the related unnecessary class fields and the test corresponding to it